### PR TITLE
Granite State + Simplified Arm Code + Driver Assist

### PR DESCRIPTION
**Pull request by BrownGenius**
I added a ArmToPosition() command that is much simpler and replaces ArmToTop, ArmToMiddle, ArmToEtc.

### CODE REVIEW

File: src/main/java/bhs/devilbotz/subsystems/stubs/ArmStub.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  public void periodic() {
+    switch (state) {
+      case MOVING_UP:
+        position += speed;
+        System.out.println("Arm Up " + position);
+        break;
+
+      case MOVING_DOWN:
+        position -= speed;
+        System.out.println("Arm Down " + position);
```
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/subsystems/stubs/ArmStub.java:32]: Remove debug prints

File: src/main/java/bhs/devilbotz/subsystems/stubs/ArmStub.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  public ArmStub() {
+    state = ArmState.STOPPED;
+  }
+
+  @Override
+  public void periodic() {
+    switch (state) {
+      case MOVING_UP:
+        position += speed;
+        System.out.println("Arm Up " + position);
```
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/subsystems/stubs/ArmStub.java:27]: Remove debug prints

File: src/main/java/bhs/devilbotz/subsystems/stubs/ArmStub.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  @Override
+  public double getPosition() {
+    return position;
+  }
+
+  @Override
+  protected void resetPosition() {
+    position = 0;
+  }
+}
```
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/subsystems/stubs/ArmStub.java:85]: Javadocs

**Review by ParkerMeyers — APPROVED** [review]
Looks good, we should test on the robot when we have time, I would just add comments basicly everywhere, just to explain variables and what some things do. Also javadocs on all methods please.

**Review by loafdog — DISMISSED** [review]
Lots of changes here... I can see how they are helpful, cut down on extra code.  But it is kind of late to be making so many changes.  If you tested this on the real robot yesterday then I'm ok with this PR being merged.

**Comment by BrownGenius** [comment]
@loafdog I agree. Unfortunately, we weren't able to test the robot last night at all because electrical issues.  This change primarily should fix the MoveToXXX commands and make them more reliable.  However, I don't think we will be able to use that functionality today.  I think Adi will just have to set the arm height manually tonight.  So, yes, unless we get to test this on a robot, we should wait before pulling it into main.

**Comment by BrownGenius** [comment]
I merged the granite_state branch into the simplified arm stub branch.  I verified that the arm/gripper controls work as expected in simulation.  I created a testing checklist that we should use as a guide for testing functionality here: https://github.com/BHSRobotix/ChargedUp2023/blob/7cdba9a73bb78bb3a0af76e18ab89b356b6bb8b3/README.md#testing-checklist

We should verify this branch on the competition bot **before** pulling into main.

**Comment by BrownGenius** [comment]
image
I added the Driver Assist functions and tested in simulation using the following shuffleboard tab

**Comment by BrownGenius** [comment]
I also added a raw hardware shuffleboard tab for HW debug: 
image

**Pull request merged**